interp/numTok.ml

```diff
@@ -201,12 +225,12 @@ struct
 
   let to_bigint_fraction (s, { int; frac; exp }) =
     let s = match s with SPlus -> "" | SMinus -> "-" in
-    let i = Bigint.of_string (s ^ int ^ frac) in
+    let i = Bigint.of_string (s ^ UnsignedNat.to_string (int ^ frac)) in
     let e =
       let e = if exp = "" then Bigint.zero else match exp.[1] with
-      | '+' -> Bigint.of_string (String.sub exp 2 (String.length exp - 2))
-      | '-' -> Bigint.(neg (of_string (String.sub exp 2 (String.length exp - 2))))
-      | _ -> Bigint.of_string (String.sub exp 1 (String.length exp - 1)) in
+      | '+' -> Bigint.of_string (UnsignedNat.to_string (String.sub exp 2 (String.length exp - 2)))
+      | '-' -> Bigint.(neg (of_string (UnsignedNat.to_string (String.sub exp 2 (String.length exp - 2)))))
+      | _ -> Bigint.of_string (UnsignedNat.to_string (String.sub exp 1 (String.length exp - 1))) in
       Bigint.(sub e (of_int (String.length frac))) in
```

Comment:
Don't we have to remove potential `_` from `frac` (but not leading zeros)? `UnsignedNat.to_string frac` should do the job.

Reply:
Indeed, this is subtle (and was wrong before). The length needs to be done once `_` are removed.

Here is a bug of 8.11:
```
Require Import Reals.
Check 1.5_%R.
(* 15e-2%R *)
Check 1.5_e2%R.
(* 15%R *)
```

Incidentally, do we really want that numbers with decimal part are printed with `e` in `R` rather than with the decimal part?

Reply:
Damn it, it was introduced in 8.10 and I'm the one to blame.

Printing in `R` could indeed be nicer. I'm adding that on top of my TODO list.